### PR TITLE
Upload support on android

### PR DESCRIPTION
**Pull request by jeluard**
Added support for upload on android (now limited to opening gallery)

### CODE REVIEW

File: android/src/main/java/com/github/alinz/reactnativewebviewbridge/WebViewBridgeManager.java [hunk 1]
```diff
@@ -67,11 +69,13 @@
     private OkHttpClient httpClient;
     private static boolean debug;
     Function<String, String> callRPC;
+    private WebViewBridgePackage aPackage;
```
Comment (yenda) [inline android/src/main/java/com/github/alinz/reactnativewebviewbridge/WebViewBridgeManager.java]: why call it `aPackage` ?

**Review by yenda — APPROVED** [review]

File: android/src/main/java/com/github/alinz/reactnativewebviewbridge/WebViewBridgeManager.java [hunk 2]
```diff
@@ -105,6 +109,28 @@ public String getName() {
 
         String origin;
         GeolocationPermissions.Callback callback;
+        WebViewBridgePackage aPackage;
+
+        public ReactWebChromeClient(WebViewBridgePackage p) {
+            this.aPackage = p;
```
Comment (mandrigin) [inline android/src/main/java/com/github/alinz/reactnativewebviewbridge/WebViewBridgeManager.java]: `this.pkg = pkg`

File: android/src/main/java/com/github/alinz/reactnativewebviewbridge/WebViewBridgeManager.java [hunk 3]
```diff
@@ -105,6 +109,28 @@ public String getName() {
...
+        }
+
+        public boolean onShowFileChooser (WebView webView, ValueCallback<Uri[]> filePathCallback, WebChromeClient.FileChooserParams fileChooserParams) {
+            aPackage.getModule().mUploadCallbackAboveL = filePathCallback;
```
Comment (mandrigin) [inline android/src/main/java/com/github/alinz/reactnativewebviewbridge/WebViewBridgeManager.java]: use a setter here, direct assignments aren't cool to debug later on

**Review by flexsurfer — APPROVED** [review]

File: android/src/main/java/com/github/alinz/reactnativewebviewbridge/WebViewBridgeManager.java [hunk 4]
```diff
@@ -67,11 +69,13 @@
     private OkHttpClient httpClient;
     private static boolean debug;
     Function<String, String> callRPC;
+    private WebViewBridgePackage aPackage;
 
-    public WebViewBridgeManager(ReactApplicationContext context, boolean debug, Function<String, String> callRPC) {
+    public WebViewBridgeManager(ReactApplicationContext context, boolean debug, Function<String, String> callRPC, WebViewBridgePackage aPackage) {
```
Comment (mandrigin) [inline android/src/main/java/com/github/alinz/reactnativewebviewbridge/WebViewBridgeManager.java]: `aPackage` -> `pkg`

File: android/src/main/java/com/github/alinz/reactnativewebviewbridge/WebViewBridgeManager.java (hunk 1 above)
Comment (jeluard) [reply android/src/main/java/com/github/alinz/reactnativewebviewbridge/WebViewBridgeManager.java]: Just copy/pasta

File: android/src/main/java/com/github/alinz/reactnativewebviewbridge/WebViewBridgePackage.java [hunk 5]
```diff
@@ -23,17 +24,25 @@ public WebViewBridgePackage(boolean debug, Function<String, String> callRPC) {
 
     @Override
     public List<NativeModule> createNativeModules(ReactApplicationContext reactApplicationContext) {
-        return new ArrayList<>();
+        List<NativeModule> modules = new ArrayList<>();
+        module = new WebViewModule(reactApplicationContext);
+        modules.add(module);
+        return modules;
```
Comment (mandrigin) [inline android/src/main/java/com/github/alinz/reactnativewebviewbridge/WebViewBridgePackage.java:30]: perhaps use `Arrays.<NativeModule>asList(module)`, like in the next method?

File: android/src/main/java/com/github/alinz/reactnativewebviewbridge/WebViewBridgeManager.java (hunk 1 above)
Comment (jeluard) [reply android/src/main/java/com/github/alinz/reactnativewebviewbridge/WebViewBridgeManager.java]: package is a keyword in Java

File: android/src/main/java/com/github/alinz/reactnativewebviewbridge/WebViewModule.java [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+import android.content.Intent;
+import android.net.Uri;
+import android.os.Build;
+import android.webkit.ValueCallback;
+import android.util.Log;
+
+import com.facebook.react.common.annotations.VisibleForTesting;
+
+public class WebViewModule extends ReactContextBaseJavaModule implements ActivityEventListener {
+    public ValueCallback<Uri> mUploadMessage;
```
Comment (mandrigin) [inline android/src/main/java/com/github/alinz/reactnativewebviewbridge/WebViewModule.java]: we don't do `m` prefixes in this codebase.
naming is also confusing. why is it a Message of type Callback?

Comment (mandrigin) [reply android/src/main/java/com/github/alinz/reactnativewebviewbridge/WebViewModule.java]: also should be private

File: android/src/main/java/com/github/alinz/reactnativewebviewbridge/WebViewModule.java [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+    public ValueCallback<Uri[]> mUploadCallbackAboveL;
```
Comment (mandrigin) [inline android/src/main/java/com/github/alinz/reactnativewebviewbridge/WebViewModule.java]: should be private
should not have `m` prefix
and we don't support even Lolipop, so `AboveL` seems pointless

File: android/src/main/java/com/github/alinz/reactnativewebviewbridge/WebViewModule.java [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+        return getCurrentActivity();
+    }
+
+    @Override
+    public void onActivityResult(Activity activity, int requestCode, int resultCode, Intent data) {
+        if (requestCode == 1) {
+            if (null == mUploadMessage && null == mUploadCallbackAboveL){
+                return;
+            }
+            Uri result = data == null || resultCode != Activity.RESULT_OK ? null : data.getData();
```
Comment (mandrigin) [inline android/src/main/java/com/github/alinz/reactnativewebviewbridge/WebViewModule.java]: this seems unused

File: android/src/main/java/com/github/alinz/reactnativewebviewbridge/WebViewModule.java [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+            if (mUploadCallbackAboveL != null) {
```
Comment (mandrigin) [inline android/src/main/java/com/github/alinz/reactnativewebviewbridge/WebViewModule.java]: why check there if `onActivityResultAboveL` also checks it?
why having `onActivityResultAboveL` at all?

**Review by mandrigin — CHANGES_REQUESTED** [review]
some changes + we need to forbid building it for anything below android M, that will save us some time and compiler warnings

File: android/src/main/java/com/github/alinz/reactnativewebviewbridge/WebViewBridgeManager.java (hunk 1 above)
Comment (mandrigin) [reply android/src/main/java/com/github/alinz/reactnativewebviewbridge/WebViewBridgeManager.java]: just call it `pkg` :)

**Comment by jeluard** [comment]
@mandrigin Addressed your comments

File: android/src/main/java/com/github/alinz/reactnativewebviewbridge/WebViewModule.java [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+import android.content.Intent;
+import android.net.Uri;
+import android.os.Build;
+import android.webkit.ValueCallback;
+import android.util.Log;
+
+import com.facebook.react.common.annotations.VisibleForTesting;
+
+public class WebViewModule extends ReactContextBaseJavaModule implements ActivityEventListener {
+    public ValueCallback<Uri[]> callback;
```
Comment (mandrigin) [inline android/src/main/java/com/github/alinz/reactnativewebviewbridge/WebViewModule.java:19]: Let’s make it private and have a public setter method

**Review by mandrigin — APPROVED** [review]
2 minor things left, but much better now!

**Pull request merged**